pkg/sql/importer/read_import_workload.go

```diff
@@ -186,7 +186,12 @@ func (w *workloadReader) readFiles(
 	}
 
 	for _, wc := range wcs {
-		if err := ctxgroup.GroupWorkers(ctx, w.parallelism, func(ctx context.Context, _ int) error {
+		if err := ctxgroup.GroupWorkers(ctx, w.parallelism, func(ctx context.Context, _ int) (retErr error) {
+			defer func() {
+				if r := recover(); r != nil {
+					retErr = errors.CombineErrors(errors.AssertionFailedf("panic occurred in workload: %v", r), retErr)
```

Comment:
nit: use %+v so that we get the stack trace of the inner error if the panic is a crdb error.

Reply:
Hm, I suppose maybe better yet would be to just keep it an `error` if it already is one?

Reply:
I have a utility similar to this in my hobby projects:

```
func Recover(do func() error) (err error) {
   defer func() {
       pValue := recover()
       if pValue != nil { 
          return
       } else if pErr, ok := pValue.(error) { 
          err = errors.Wrap(err, "panic'd error")
       } else {
          err = errors.AssertionFailedF("panic: %v", pValue)
       }
   }()
   return do()
}
```

Then code like this becomes:
```
ctxgroup.groupWorkers(..., func(ctx context.Context, _ int) error {
   return Recover(func() error {
      // Write code as normal
   })
})
```

It is important to wrap the error again because you want the stack trace to indicate where the error was thrown and where the error was recovered.

Reply:
But mostly this is part of my long running campaign to get rid of use cases for non-trivial defers.